hoot-core/src/main/cpp/hoot/core/io/OsmJsonReader.cpp

```diff
@@ -0,0 +1,410 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "OsmJsonReader.h"
+
+// hoot
+#include <hoot/core/Hoot.h>
+#include <hoot/core/util/HootException.h>
+#include <hoot/core/util/Log.h>
+#include <hoot/core/Factory.h>
+
+// Boost
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+namespace pt = boost::property_tree;
+
+// Qt
+#include <QTextCodec>
+#include <QEventLoop>
+#include <QTextStream>
+#include <QtNetwork/QNetworkReply>
+#include <QtNetwork/QNetworkRequest>
+#include <QtNetwork/QNetworkAccessManager>
+
+// Standard
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <unistd.h>
+using namespace std;
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(OsmMapReader, OsmJsonReader)
+
+// Default constructor
+OsmJsonReader::OsmJsonReader():
+  _defaultStatus(Status::Invalid),
+  _useDataSourceIds(true),
+  _defaultCircErr(15.0),
+  _propTree(),
+  _version(""),
+  _generator(""),
+  _timestamp_base(""),
+  _copyright(""),
+  _url(""),
+  _isFile(false),
+  _isWeb(false)
+{
+  // Do nothing special
+}
+
+OsmJsonReader::~OsmJsonReader()
+{
+  close();
+}
+
+bool OsmJsonReader::isSupported(QString url)
```

Comment:
This should be a bit more restrictive. The OsmMapReaderFactory will use this method to determine which reader should be used for opening a file.

https://github.com/ngageoint/hootenanny/blob/2ef291df533d113a319625c2d13ea1c1a129c949/hoot-core/src/main/cpp/hoot/core/io/OsmMapReaderFactory.cpp#L102

The local file check should likely check for a `.json` extension.

I'm not sure about the best way to handle the web request. I think this is the only reader that supports http:// so your http/https check is probably good enough for now.